src/plugins/intel_gpu/src/plugin/ops/constant.cpp

```diff
@@ -206,6 +208,14 @@ static void CreateConstantOp(ProgramBuilder& p, const std::shared_ptr<ov::op::v0
             }
         } else if (ov::is_type<ov::op::v3::ROIAlign>(outOp) || ov::is_type<ov::op::v9::ROIAlign>(outOp)) {
             consts[op].needsBatchInterpretation = constDims.size() == 1;
+        } else if ((ov::is_type<ov::op::v5::Loop>(outOp) || ov::is_type<ov::op::v0::TensorIterator>(outOp)) && constDims.size() == 1) {
+            // when inner network has 1d parameter which is connected to outer loop's constant 1d data,
+            // outer constant 1d data and inner 1d parameter has same bytes_count but layout is different
+            // (outer constant is [1, N, 1, 1] but inner parameter is [N, 1, 1, 1]).
+            // To pass check_memory_to_set in input_layout::set_data for this case, Set constDims to [N, 1, 1, 1]
+            // when constDims is one dim and user op is Loop or TensorIterator.
+            for (size_t i = 0; i < 3; i++)
+                constDims.push_back(1);
```

Comment:
consts[op].needsBatchInterpretation will do this.

Reply:
Fixed it. thanks.